From c78819245ff8a270f97c9f800773e727918be838 Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Fri, 15 Jul 2016 11:12:35 -0400
Subject: [PATCH] Add gettext support for .policy files

gettext can extract strings from and merge them back into xml
file formats, with the help of .its files.

https://bugs.freedesktop.org/show_bug.cgi?id=96940

Applied-Upstream: 0.114
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=863207

---
 data/Makefile.am | 5 +++++
 data/polkit.its  | 7 +++++++
 data/polkit.loc  | 6 ++++++
 3 files changed, 18 insertions(+)
 create mode 100644 data/polkit.its
 create mode 100644 data/polkit.loc

diff --git a/data/Makefile.am b/data/Makefile.am
index fe0f1d5..18693fe 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -36,6 +36,11 @@ pkgconfig_DATA = polkit-gobject-1.pc polkit-agent-1.pc
 
 # ----------------------------------------------------------------------------------------------------
 
+itsdir = $(datadir)/gettext/its
+its_DATA = polkit.loc polkit.its
+
+# ----------------------------------------------------------------------------------------------------
+
 systemdservice_in_files = polkit.service.in
 
 if HAVE_SYSTEMD
diff --git a/data/polkit.its b/data/polkit.its
new file mode 100644
index 0000000..1312ecb
--- /dev/null
+++ b/data/polkit.its
@@ -0,0 +1,7 @@
+<?xml version="1.0"?>
+<its:rules xmlns:its="http://www.w3.org/2005/11/its"
+           version="2.0">
+  <its:translateRule selector="/action/description |
+                               /action/message"
+                     translate="yes"/>
+</its:rules>
diff --git a/data/polkit.loc b/data/polkit.loc
new file mode 100644
index 0000000..c7427ec
--- /dev/null
+++ b/data/polkit.loc
@@ -0,0 +1,6 @@
+<?xml version="1.0"?>
+<locatingRules>
+  <locatingRule name="polkit policy" pattern="*.policy">
+    <documentRule localName="policyconfig" target="polkit.its"/>
+  </locatingRule>
+</locatingRules>
-- 
2.11.0

From 32e9a69c335324a53a2c0ba4e0b513fb044be0fd Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Thu, 20 Oct 2016 10:50:58 +1000
Subject: [PATCH] gettext: switch to default-translate "no"

The default appears to be to translate all entries. This rule never takes
effect, the path to /action/message and /action/description is wrong (/action
is not a root node). Since we wanted them to be translated, it doesn't matter.

But it also translates all other tags (vendor, allow_any, etc.) and that
causes polkit to be unhappy, it can't handle the various language versions of
"no"

** (polkitd:27434): WARNING **: Unknown PolkitImplicitAuthorization string
'tidak'

Switch to a default of "no" and explicitly include the message and description
strings to be translated.

The patch was modified for PolicyKit by Ondrej Holy <oholy@redhat.com>.

https://bugs.freedesktop.org/show_bug.cgi?id=98366
---
 data/polkit.its | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/data/polkit.its b/data/polkit.its
index 1312ecb..1c37e6b 100644
--- a/data/polkit.its
+++ b/data/polkit.its
@@ -1,7 +1,8 @@
 <?xml version="1.0"?>
 <its:rules xmlns:its="http://www.w3.org/2005/11/its"
            version="2.0">
-  <its:translateRule selector="/action/description |
-                               /action/message"
+  <its:translateRule selector="//*" translate="no"/>
+  <its:translateRule selector="//action/description |
+                               //action/message"
                      translate="yes"/>
 </its:rules>
-- 
2.11.0

